pkg/blobfuse-proxy/init.sh

```diff
@@ -17,21 +17,54 @@
 set -xe
 
 INSTALL_BLOBFUSE_PROXY=${INSTALL_BLOBFUSE_PROXY:-true}
-INSTALL_BLOBFUSE=${INSTALL_BLOBFUSE:-true}
 DISABLE_UPDATEDB=${DISABLE_UPDATEDB:-true}
 SET_MAX_OPEN_FILE_NUM=${SET_MAX_OPEN_FILE_NUM:-true}
 SET_READ_AHEAD_SIZE=${SET_READ_AHEAD_SIZE:-true}
 READ_AHEAD_KB=${READ_AHEAD_KB:-15380}
 
 HOST_CMD="nsenter --mount=/proc/1/ns/mnt"
 
-# install/update blobfuse
-if [ "${INSTALL_BLOBFUSE}" = "true" ]
+if [ "${INSTALL_BLOBFUSE}" = "true" ] || [ "${INSTALL_BLOBFUSE2}" = "true" ]
 then
   cp /blobfuse-proxy/packages-microsoft-prod.deb /host/etc/
-  yes | $HOST_CMD dpkg -i /etc/packages-microsoft-prod.deb && \
-  $HOST_CMD apt update && \
-  $HOST_CMD apt-get install -y fuse blobfuse2 blobfuse="${BLOBFUSE_VERSION}" && \
+  # when running dpkg -i /etc/packages-microsoft-prod.deb, need to enter y to continue. 
+  # refer to https://stackoverflow.com/questions/45349571/how-to-install-deb-with-dpkg-non-interactively
+  yes | $HOST_CMD dpkg -i /etc/packages-microsoft-prod.deb && $HOST_CMD apt update
+
+  pkg_list=""
+  if [ "${INSTALL_BLOBFUSE}" = "true" ]
+  then
+    pkg_list="${pkg_list} fuse"
+    # install blobfuse with latest version or specific version
+    if [ -z "${BLOBFUSE_VERSION}" ]; then
+      echo "install blobfuse with latest version"
+      pkg_list="${pkg_list} blobfuse"
+    else
+      pkg_list="${pkg_list} blobfuse=${BLOBFUSE_VERSION}"
+    fi
+  fi
+
+  if [ "${INSTALL_BLOBFUSE2}" = "true" ]
+  then
+    release=$($HOST_CMD lsb_release -rs)
+    if [ "$release" = "18.04" ]; then
+      echo "install fuse for blobfuse2"
+      pkg_list="${pkg_list} fuse"
+    else
+      echo "install fuse3 for blobfuse2, current release is $release"
+      pkg_list="${pkg_list} fuse3"
+    fi
+
+    # install blobfuse2 with latest version or specific version
+    if [ -z "${BLOBFUSE2_VERSION}" ]; then
+      echo "install blobfuse2 with latest version"
+      pkg_list="${pkg_list} blobfuse2"
+    else
+      pkg_list="${pkg_list} blobfuse2=${BLOBFUSE2_VERSION}"
+    fi
+  fi
+  echo "begin to install ${pkg_list}"
+  $HOST_CMD apt-get install -y $pkg_list
```

Comment:
We are using fixed version of deb package: https://github.com/kubernetes-sigs/blob-csi-driver/blob/master/pkg/blobplugin/Dockerfile#L36-L37. Is it possible that some of the blobfuse/blobfuse2 version are not included in it?

Reply:
yes, that's possible if `packages-microsoft-prod.deb` in the driver is old, and since BLOBFUSE_VERSION can be specified by end user, if it's not supported, then it would return error, so that's ok